test/integration/appsource/cymbal_bank_test.go

```diff
@@ -175,11 +175,18 @@ func TestSourceCymbalBank(t *testing.T) {
 				lastCommit := gitApp.GetLatestCommit()
 				// filter builds triggered based on pushed commit sha
 				buildListCmd := fmt.Sprintf("builds list --region=%s --filter substitutions.COMMIT_SHA='%s' --project %s", region, lastCommit, servicesInfoMap[serviceName].ProjectID)
+				retriesBuildTrigger := 0
 				// poll build until complete
 				pollCloudBuild := func(cmd string) func() (bool, error) {
 					return func() (bool, error) {
 						build := gcloud.Runf(t, cmd).Array()
 						if len(build) < 1 {
+							if retriesBuildTrigger == 3 {
```

Comment:
Idea: if you start `retriesBuildTrigger` at `1` then you can `retriesBuildTrigger % 3 == 0` to push every 3 runs, etc.